corehq/apps/app_manager/app_translations/generators.py

```diff
@@ -185,6 +186,29 @@ def _filter_invalid_rows_for_form(self, rows, form_id, label_index):
                 valid_rows.append(row)
         return valid_rows
 
+    @cached_property
+    def _blacklisted_translations(self):
+        return TransifexBlacklist.objects.filter(domain=self.domain, app_id=self.app_id).all()
+
+    def _filter_invalid_rows_for_module(self, rows, module_id, case_property_index, list_or_detail_index):
```

Comment:
Do you plan to refine this logic? 
There are quite some iterations here.

I was thinking of
1. fetch the blacklisted translations by module_id instead, which thus result in more queries
2. when you get the blacklisted translations for the app, iterate over them to create a dict of blacklisted translations per module.

then later you could just do
```
blacklisted_case_properties = self. _blacklisted_translations[module_id]
for i, row in enumerate(rows):
   if row[case_property_index] not in blacklisted_case_properties:
       valid_rows.append(row)
```